test/common/http/header_utility_test.cc

```diff
@@ -201,7 +201,7 @@ name: test-header
   EXPECT_EQ(HeaderUtility::HeaderMatchType::Present, header_data.header_match_type_);
 }
 
-TEST(HeaderDataConstructorTest, ExactMatchSpecifier) {
+TEST(HeaderDataConstructorTest, DEPRECATED_FEATURE_TEST(ExactMatchSpecifier)) {
```

Comment:
I guess I'm still vaguely curious why this was passing CI (with compile_time_options rejecting deprecated fields)?

Reply:
I'm not sure, by looking at the code, `parseHeaderMatcherFromYaml()` calls the `TestUtility::loadFromYaml(yaml, header_matcher)` which seems essentially called the [MessageUtil::loadFromYaml()](https://github.com/envoyproxy/envoy/blob/fa754faf339d9d9180934af710ede614355b48b3/source/common/protobuf/utility.cc#L403-L413) that does not check for the deprecated fields?